### DISCUSSION TITLE
Which test hangs?

### DISCUSSION THREAD

**Discussion by donnyxray** [pestphp/pest#864] [General]
I have a test that hangs. Indefinitely. But I don't know which one.

Are any of these strategies possible?

* Enable a timeout. Any test that takes more than N seconds should be killed, and an error shown.
* Get Pest to display which test is running _prior_ to running it. Currently you have no idea what pest is doing until _after_ it has the results.
* Get Pest to catch my CTRL+C and display which test I just aborted.

Other suggestions welcome!

**Comment by JonPurvis** [comment]
Hey @donnyxray 👋 

How are you running tests? Are you executing:

```bash
./vendor/bin/pest
```

That should show the file it's getting to and even the test before the one it seems to hang on? 

You could also make use of [`todo()`](https://pestphp.com/docs/skipping-tests#content-creating-todos) and mark your tests with it one by one, until your test suite stops hanging, then you'll know which one is causing the hang. 

Once you've solved that, it might be worth running 

```bash
./vendor/bin/pest --profile
```

So you can see any other tests that are running for a while. That won't work unless you're suite finishes though, so you'll need to find the hanging test first.
